sklearn/manifold/tests/test_t_sne.py

```diff
@@ -280,13 +288,16 @@ def test_fit_csr_matrix():
 def test_preserve_trustworthiness_approximately_with_precomputed_distances():
     # Nearest neighbors should be preserved approximately.
     random_state = check_random_state(0)
-    X = random_state.randn(100, 2)
-    D = squareform(pdist(X), "sqeuclidean")
-    tsne = TSNE(n_components=2, perplexity=2, learning_rate=100.0,
-                metric="precomputed", random_state=0, verbose=0)
-    X_embedded = tsne.fit_transform(D)
-    assert_almost_equal(trustworthiness(D, X_embedded, n_neighbors=1,
-                                        precomputed=True), 1.0, decimal=1)
+    for i in range(5):
+        X = random_state.randn(100, 2)
+        D = squareform(pdist(X), "sqeuclidean")
+        tsne = TSNE(n_components=2, perplexity=2, learning_rate=100.0,
+                    early_exaggeration=2.0, metric="precomputed",
+                    random_state=i, verbose=0, neighbors_method='brute')
+        X_embedded = tsne.fit_transform(D)
+        t = trustworthiness(D, X_embedded, n_neighbors=1,
+                            precomputed=True)
+        assert t > .95
```

Comment:
`pytest --durations=1 sklearn/manifold/tests/test_t_sne.py` tells me that this test take more than 1s (1.68s) It would be great to make it run faster, maybe we with 3 random datasets instead of 5.